engine/src/main/java/org/hibernate/validator/spi/property/PropertyAccessorSelector.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.spi.property;
+
+import java.lang.reflect.Method;
+
+/**
+ * Allows to customize property naming, selection and retrieval based on reflection metadata.
+ *
+ * @since 6.1
+ */
+public interface PropertyAccessorSelector {
+
+	/**
+	 * Derives property name from a method. If it is not possible will return {@code null}.
+	 *
+	 * TODO: should this support non-getters methods as well (like setters, void return type etc.) ?
+	 *
+	 * <p>Example for classic java bean convention: {@code getFoo()} -> {@code foo}</p>
+	 *
+	 * @param method method to be analyzed
+	 * @return property name which is exposed by this method, {@code null} if method is unknown to current
+	 * selector instance.
+	 */
+	String getPropertyName(Method method);
```

Comment:
Hi Gunnar,

I'll take a look. 

How `Property` is different from `ConstrainedElement` in this case ?